data_store/controllers/notify.py

```diff
@@ -0,0 +1,85 @@
+from datetime import datetime
+
+from notifications_python_client.notifications import NotificationsAPIClient
+
+from config import Config
+from data_store.const import FUND_TYPE_ID_TO_FRIENDLY_NAME
+from data_store.controllers.load_functions import get_or_generate_submission_id
+from data_store.controllers.retrieve_submission_file import get_custom_file_name
+from data_store.db.queries import get_programme_by_id_and_round
+
+
+def send_email(
+    email_address: str,
+    template_id: str,
+    notify_key: str | None,
+    personalisation: dict,
+) -> None:
+    notify_client = NotificationsAPIClient(notify_key)
+
+    notify_client.send_email_notification(
+        email_address=email_address,
+        template_id=template_id,
+        personalisation=personalisation,
+    )
+
+
+def send_la_confirmation_emails(
+    filename: str,
+    user_email: str,
+    fund_name: str,
+    current_reporting_period: str,
+    programme_name: str,
+    fund_type: str,
+):
+    if not (programme_name or fund_type):
+        raise ValueError("Cannot personalise confirmation email without programme name or fund type")
+
+    send_email(
+        email_address=user_email,
+        template_id=Config.LA_CONFIRMATION_EMAIL_TEMPLATE_ID,
+        notify_key=Config.NOTIFY_API_KEY,
+        personalisation={
+            "name_of_fund": fund_name,
+            "reporting_period": current_reporting_period,
+            "filename": filename,
+            "place_name": programme_name or "",
+            "fund_type": FUND_TYPE_ID_TO_FRIENDLY_NAME.get(fund_type, ""),
+            "date_of_submission": datetime.now().strftime("%e %B %Y").strip(),
+        },
+    )
+
+
+def send_fund_confirmation_email(
+    fund_name: str,
+    fund_email: str,
+    round_number: int,
+    programme_name: str,
+    fund_type: str,
+    programme_id: str,
+):
+    if not (programme_name or fund_type):
+        raise ValueError("Cannot personalise confirmation email without programme name or fund type")
+
+    programme = get_programme_by_id_and_round(programme_id, round_number)
+    _, submission_id = get_or_generate_submission_id(programme, round_number, fund_type)
+
+    if submission_id is None:
+        raise ValueError("Submission ID not found")
+
+    object_name = f"{fund_type}/{submission_id}"
+    link_to_file = f"{Config.FIND_SERVICE_BASE_URL}/retrieve-spreadsheet/{object_name}"
```

Comment:
This is the only I am not too sure about..

Using the `url_for` function would create an Flask `app.context` dependency inside `notify.py`. Which I think we should try to avoid.

Passing the URL as a parameter to the function is also not as simple because I don't have `submission.id` outside of my `notify.py` function.

Happy to discuss it tomorrow.

I see we use `app.logger` and `g` inside `data-store` directory (f.x. `ingest.py`). So maybe it's not a big deal. In my mind app context should not leave `find` and `submit`.

Reply:
Hmm you make a good point...let's chat indeed

Reply:
Using `url_for` now.